src/main/native/smithwaterman/PairWiseSW.h

```diff
@@ -407,11 +408,14 @@ void inline getCIGAR(SeqPair *p, int16_t *cigarBuf_, int32_t tid)
                 state = 'R';
                 break;
         }
-	// 4 is u16 + Char + Null; look into how to handle the error state  
-        if (curSize < (maxSize - 4)) 
-		curSize += snprintf(p->cigar + curSize, 4, "%d%c", cigarArray[2 * i + 1], state);
+	// expectedLength for converting int to str w/ extra padding for '\0'
+	int expectedLength = snprintf( NULL, 0, "%d%c", cigarArray[2 * i + 1], state) + 1;
+	if (curSize >= 0 && expectedLength > 1)
```

Comment:
Checking that append only when expetced length is calculated correctly. 
In case of overflow curSize can go negative, during iteration but highly unlikely. Can add extra condition as follows :
`if (curSize >= 0 && expectedLength > 1 && (curSize < INT_MAX - expectedLength))`